src/urllib3/response.py

```diff
@@ -768,7 +796,14 @@ def _fp_read(self, amt: int | None = None) -> bytes:
         """
         assert self._fp
         c_int_max = 2**31 - 1
-        if (
+        if one_read:
+            # StringIO doesn't like amt=None
+            if amt is None:
+                return self._fp.read1()
+            if amt > c_int_max and (util.IS_PYOPENSSL or sys.version_info < (3, 10)):
+                return self._fp.read1(c_int_max)
```

Comment:
I believe there is a possibility of an overflow error here because the `(self.length_remaining and self.length_remaining > c_int_max)` part of the following condition is skipped for `one_read`.
https://github.com/urllib3/urllib3/blob/4dd4a824e60ba3bb81f319cf2b8e320aa8a87dc6/src/urllib3/response.py#L771-L774

When `amt` is `None` and the content length is known, `http.client.HTTPResponse.read1` will [try to read](https://github.com/python/cpython/blob/9302f05f9af07332c414b3c19003efd1b1763cf3/Lib/http/client.py#L654-L655) the length amount of bytes. This can lead to an overflow error as described in the docstring of `_fp_read`.

Ideally (and probably required for 100% coverage), a test similar to existing `test_requesting_large_resources_via_ssl` has to check possible cases for `read1`. The problem with the existing test is that it causes timeouts in CI. Adding a similar one will worsen the situation. Although, a test for `read1` should be quicker because it will not read all 2 GiB.